src/main/java/com/google/devtools/build/lib/analysis/BuildView.java

```diff
@@ -454,13 +462,23 @@ private AnalysisResult createResult(
       AnalysisOptions viewOptions,
       SkyframeAnalysisResult skyframeAnalysisResult,
       Set<ConfiguredTarget> targetsToSkip,
+      List<String> requestedTargets,
       TopLevelTargetsAndConfigsResult topLevelTargetsWithConfigs)
       throws InterruptedException {
     Set<Label> testsToRun = loadingResult.getTestsToRunLabels();
     Set<ConfiguredTarget> configuredTargets =
         Sets.newLinkedHashSet(skyframeAnalysisResult.getConfiguredTargets());
     ImmutableSet<AspectValue> aspects = ImmutableSet.copyOf(skyframeAnalysisResult.getAspects());
 
+    // Targets that should be skipped and were requested on the command line
+    // will be marked as failed.
+    ImmutableSet.Builder<ConfiguredTarget> failedTargets = ImmutableSet.builder();
```

Comment:
Hmm. I vaguely remember not doing that because it meant duplicating some code from here into TopLevelConstraintSemantics, but I should re-visit it. What you're suggesting sounds reasonable to me. I'll give that a try.